### PR TITLE
[Profiling] Raise max stack depth to 512

### PR DESCRIPTION
**Pull request by morrisonlevi**
### Description

Some customers are seeing truncated stacks. This PR allows going up
to 512 frames deep now but it no longer allocates the max stack depth
when initializing samples buffer. This avoids waste when the stack is
much smaller than the limit, which it very often will be with a higher
limit.

### Readiness checklist
- [x] Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.

### CODE REVIEW

**Review by r1viollet — DISMISSED** [review]
LGTM

File: profiling/src/profiling.rs [hunk 1]
```diff
@@ -266,9 +266,8 @@ unsafe fn extract_line_no(execute_data: &zend_execute_data) -> u32 {
 unsafe fn collect_stack_sample(
     top_execute_data: *mut zend_execute_data,
 ) -> Result<Vec<ZendFrame>, Utf8Error> {
-    let mut samples = Vec::new();
-    let max_depth = 128;
-    samples.reserve(max_depth); // todo: try_reserve
+    let max_depth = 512;
+    let mut samples = Vec::with_capacity(max_depth >> 3);
```
Comment (r1viollet) [inline profiling/src/profiling.rs:270]: Minor: We could explicitly write the reserve size

Comment (morrisonlevi) [reply profiling/src/profiling.rs:270]: We could, but then we should probably assert it is less than `max_depth`. Didn't seem worth it.

**Review by bwoebi — APPROVED** [review]
Re-ack

**Pull request merged**